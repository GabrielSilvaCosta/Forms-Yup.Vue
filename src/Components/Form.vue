<template>
    <form
      class="form max-w-md mx-auto mt-8 p-4 bg-white shadow-md rounded"
      @submit.prevent="submitForm"
    >
      <div class="form-group">
        <label for="name" class="text-lg font-semibold">Nome:</label>
        <input
          type="text"
          v-model="form.name"
          class="mt-1 p-2 border rounded focus:outline-none focus:ring focus:border-blue-300"
        />
        <span v-if="errors.name" class="error text-sm">{{ errors.name[0] }}</span>
      </div>
      <div class="form-group mt-6">
        <label for="email" class="text-lg font-semibold">Email:</label>
        <input
          type="email"
          v-model="form.email"
          class="mt-1 p-2 border rounded focus:outline-none focus:ring focus:border-blue-300"
        />
        <span v-if="errors.email" class="error text-sm">{{ errors.email[0] }}</span>
      </div>
  
      <div class="mt-6 form-group flex items-center">
        <input
          type="checkbox"
          id="checkboxOne"
          v-model="checkboxOne"
          class="w-6 h-6 p-1 border border-gray-400 rounded focus:outline-none focus:ring focus:border-blue-300"
        />
        <label for="checkboxOne" class="ml-2 text-lg font-semibold">Aceitar Termos</label>
      </div>
  
      <div class="mt-6">
        <button
          type="submit"
          class="py-2 px-4 bg-blue-500 text-white rounded hover:bg-blue-600"
        >
          Enviar
        </button>
      </div>
    </form>
  </template>
  
  <script>
  import { ref } from 'vue';
  import formSchema from '../schemas/formSchema';
  import * as yup from 'yup'; 
  
  export default {
    setup() {
      const checkboxOne = ref(true);
  
      return {
        checkboxOne,
      };
    },
    data() {
      return {
        form: {
          name: '',
          email: '',
        },
        errors: {},
      };
    },
    methods: {
      async submitForm() {
        try {
          await formSchema.validate(this.form, { abortEarly: false });
          alert('FormulÃ¡rio enviado com sucesso!');
        } catch (err) {
          if (err instanceof yup.ValidationError) {
            const validationErrors = {};
            err.inner.forEach((error) => {
              validationErrors[error.path] = error.errors;
            });
            this.errors = validationErrors;
          }
        }
      },
    },
  };
  </script>
  
  <style scoped>
  .error {
    color: red;
  }
  </style>
  