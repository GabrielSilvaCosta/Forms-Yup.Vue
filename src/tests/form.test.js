import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";
import Form from "../Components/Form.vue";

describe("Form", () => {
  it("should render correctly", () => {
    const wrapper = mount(Form);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should submit the form", async () => {
    const wrapper = mount(Form);
    const button = wrapper.get("button");
    await button.trigger("click");
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should display errors", async () => {
    const wrapper = mount(Form);
    await wrapper.setData({ name: "Teste", email: "teste@teste" });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render an error message if the name is empty", async () => {
    const wrapper = mount(Form);
    await wrapper.setData({ name: "", email: "teste@teste" });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
