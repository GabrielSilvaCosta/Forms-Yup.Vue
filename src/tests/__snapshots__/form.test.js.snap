// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Form > should display errors 1`] = `
"<form data-v-362c0340="" class="form max-w-md mx-auto mt-8 p-4 bg-white shadow-md rounded">
  <div data-v-362c0340="" class="form-group"><label data-v-362c0340="" for="name" class="text-lg font-semibold">Nome:</label><input data-v-362c0340="" type="text" class="mt-1 p-2 border rounded focus:outline-none focus:ring focus:border-blue-300">
    <!--v-if-->
  </div>
  <div data-v-362c0340="" class="form-group mt-6"><label data-v-362c0340="" for="email" class="text-lg font-semibold">Email:</label><input data-v-362c0340="" type="email" class="mt-1 p-2 border rounded focus:outline-none focus:ring focus:border-blue-300">
    <!--v-if-->
  </div>
  <div data-v-362c0340="" class="mt-6 form-group flex items-center"><input data-v-362c0340="" type="checkbox" id="checkboxOne" class="w-6 h-6 p-1 border border-gray-400 rounded focus:outline-none focus:ring focus:border-blue-300"><label data-v-362c0340="" for="checkboxOne" class="ml-2 text-lg font-semibold">Aceitar Termos</label></div>
  <div data-v-362c0340="" class="mt-6"><button data-v-362c0340="" type="submit" class="py-2 px-4 bg-blue-500 text-white rounded hover:bg-blue-600"> Enviar </button></div>
</form>"
`;

exports[`Form > should render an error message if the name is empty 1`] = `
"<form data-v-362c0340="" class="form max-w-md mx-auto mt-8 p-4 bg-white shadow-md rounded">
  <div data-v-362c0340="" class="form-group"><label data-v-362c0340="" for="name" class="text-lg font-semibold">Nome:</label><input data-v-362c0340="" type="text" class="mt-1 p-2 border rounded focus:outline-none focus:ring focus:border-blue-300">
    <!--v-if-->
  </div>
  <div data-v-362c0340="" class="form-group mt-6"><label data-v-362c0340="" for="email" class="text-lg font-semibold">Email:</label><input data-v-362c0340="" type="email" class="mt-1 p-2 border rounded focus:outline-none focus:ring focus:border-blue-300">
    <!--v-if-->
  </div>
  <div data-v-362c0340="" class="mt-6 form-group flex items-center"><input data-v-362c0340="" type="checkbox" id="checkboxOne" class="w-6 h-6 p-1 border border-gray-400 rounded focus:outline-none focus:ring focus:border-blue-300"><label data-v-362c0340="" for="checkboxOne" class="ml-2 text-lg font-semibold">Aceitar Termos</label></div>
  <div data-v-362c0340="" class="mt-6"><button data-v-362c0340="" type="submit" class="py-2 px-4 bg-blue-500 text-white rounded hover:bg-blue-600"> Enviar </button></div>
</form>"
`;

exports[`Form > should render correctly 1`] = `
"<form data-v-362c0340="" class="form max-w-md mx-auto mt-8 p-4 bg-white shadow-md rounded">
  <div data-v-362c0340="" class="form-group"><label data-v-362c0340="" for="name" class="text-lg font-semibold">Nome:</label><input data-v-362c0340="" type="text" class="mt-1 p-2 border rounded focus:outline-none focus:ring focus:border-blue-300">
    <!--v-if-->
  </div>
  <div data-v-362c0340="" class="form-group mt-6"><label data-v-362c0340="" for="email" class="text-lg font-semibold">Email:</label><input data-v-362c0340="" type="email" class="mt-1 p-2 border rounded focus:outline-none focus:ring focus:border-blue-300">
    <!--v-if-->
  </div>
  <div data-v-362c0340="" class="mt-6 form-group flex items-center"><input data-v-362c0340="" type="checkbox" id="checkboxOne" class="w-6 h-6 p-1 border border-gray-400 rounded focus:outline-none focus:ring focus:border-blue-300"><label data-v-362c0340="" for="checkboxOne" class="ml-2 text-lg font-semibold">Aceitar Termos</label></div>
  <div data-v-362c0340="" class="mt-6"><button data-v-362c0340="" type="submit" class="py-2 px-4 bg-blue-500 text-white rounded hover:bg-blue-600"> Enviar </button></div>
</form>"
`;

exports[`Form > should submit the form 1`] = `
"<form data-v-362c0340="" class="form max-w-md mx-auto mt-8 p-4 bg-white shadow-md rounded">
  <div data-v-362c0340="" class="form-group"><label data-v-362c0340="" for="name" class="text-lg font-semibold">Nome:</label><input data-v-362c0340="" type="text" class="mt-1 p-2 border rounded focus:outline-none focus:ring focus:border-blue-300">
    <!--v-if-->
  </div>
  <div data-v-362c0340="" class="form-group mt-6"><label data-v-362c0340="" for="email" class="text-lg font-semibold">Email:</label><input data-v-362c0340="" type="email" class="mt-1 p-2 border rounded focus:outline-none focus:ring focus:border-blue-300">
    <!--v-if-->
  </div>
  <div data-v-362c0340="" class="mt-6 form-group flex items-center"><input data-v-362c0340="" type="checkbox" id="checkboxOne" class="w-6 h-6 p-1 border border-gray-400 rounded focus:outline-none focus:ring focus:border-blue-300"><label data-v-362c0340="" for="checkboxOne" class="ml-2 text-lg font-semibold">Aceitar Termos</label></div>
  <div data-v-362c0340="" class="mt-6"><button data-v-362c0340="" type="submit" class="py-2 px-4 bg-blue-500 text-white rounded hover:bg-blue-600"> Enviar </button></div>
</form>"
`;
